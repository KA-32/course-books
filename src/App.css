body {
  background-color: whitesmoke;
  font-family: "Roboto", sans-serif;
  margin: 0;
  padding: 0;
}

.main-container {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  margin: 0 auto 60px auto;
  max-width: 1280px;
}

.title {
  margin: 2em 0;
  color: #0d74d8;
}
