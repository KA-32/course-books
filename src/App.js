/**
 * Root component.
 * @component
 * Initialise view with required components.
 */
import React, { useState } from "react";

import SearchBox from "./components/SearchBox/SearchBox";
import CourseList from "./components/CourseList/CourseList";
import Autocomplete from "./components/AutoSuggestion/Autocomplete";


import CourseBooks from "./utils/coursebooks";

import "./App.css";

const App = () => {
  const [filteredBooks, setBooks] = useState([]);

  const addBook = (value) => {
    let newCourseBook = CourseBooks.getCoursebook(value);
    let booksData = [...filteredBooks];
    booksData.push(newCourseBook);
    setBooks(booksData);
  };

  return (
    <section className="main-container" role="main">
      <h1 className="title">Search Books</h1>
      {/* <SearchBox addBook={addBook} titles={CourseBooks.getAllTitles()} /> */}
      <Autocomplete addBook={addBook}/>
      <CourseList books={filteredBooks} />
    </section>
  );
};

export default App;
