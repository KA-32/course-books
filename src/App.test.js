import React from "react";
import { render } from "@testing-library/react";
import App from "./App";

test("Title Should be 'Search Books'", () => {
  const { getByText } = render(<App />);
  const linkElement = getByText(/Search Books/i);
  expect(linkElement).toBeInTheDocument();
});
