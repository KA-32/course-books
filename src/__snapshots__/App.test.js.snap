// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component App to match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="main-container"
        role="main"
      >
        <h1
          class="title"
        >
          Search Books
        </h1>
        <div
          class="form-search"
        >
          <div>
            <input
              aria-label="Search box"
              aria-required="true"
              class="search-input"
              name="book-search"
              placeholder="Enter summary of book"
              type="text"
              value=""
            />
            <button
              class="form-submit-btn"
              disabled=""
            >
              Submit
            </button>
          </div>
        </div>
        <ul
          class="books-list"
        />
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="main-container"
      role="main"
    >
      <h1
        class="title"
      >
        Search Books
      </h1>
      <div
        class="form-search"
      >
        <div>
          <input
            aria-label="Search box"
            aria-required="true"
            class="search-input"
            name="book-search"
            placeholder="Enter summary of book"
            type="text"
            value=""
          />
          <button
            class="form-submit-btn"
            disabled=""
          >
            Submit
          </button>
        </div>
      </div>
      <ul
        class="books-list"
      />
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
